core/src/main/java/org/sonarsource/sonarlint/core/VersionSoonUnsupportedHelper.java

```diff
@@ -0,0 +1,113 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core;
+
+import com.google.common.eventbus.Subscribe;
+import java.util.Set;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import org.sonarsource.sonarlint.core.clientapi.SonarLintClient;
+import org.sonarsource.sonarlint.core.clientapi.client.message.MessageType;
+import org.sonarsource.sonarlint.core.clientapi.client.message.ShowSoonUnsupportedMessageParams;
+import org.sonarsource.sonarlint.core.commons.ConnectionKind;
+import org.sonarsource.sonarlint.core.commons.Version;
+import org.sonarsource.sonarlint.core.commons.log.SonarLintLogger;
+import org.sonarsource.sonarlint.core.event.BindingConfigChangedEvent;
+import org.sonarsource.sonarlint.core.event.ConfigurationScopesAddedEvent;
+import org.sonarsource.sonarlint.core.repository.config.ConfigurationRepository;
+import org.sonarsource.sonarlint.core.repository.connection.ConnectionConfigurationRepository;
+import org.sonarsource.sonarlint.core.serverapi.system.ServerInfo;
+import org.sonarsource.sonarlint.core.serverconnection.VersionUtils;
+
+@Named
+@Singleton
+public class VersionSoonUnsupportedHelper {
+
+  private static final String UNSUPPORTED_NOTIFICATION_ID = "sonarlint.unsupported.%s.id";
+  private static final String NOTIFICATION_MESSAGE = "The version used by the current connection '%s' will be soon unsupported. " +
+    "Please consider upgrading to the latest %s LTS version to ensure continued support and access to the latest features.";
+  private static final SonarLintLogger LOG = SonarLintLogger.get();
+  private final SonarLintClient client;
+  private final ConfigurationRepository configRepository;
+  private final ConnectionConfigurationRepository connectionRepository;
+  private final ServerApiProvider serverApiProvider;
+
+  public VersionSoonUnsupportedHelper(SonarLintClient client, ConfigurationRepository configRepository,
+    ConnectionConfigurationRepository connectionRepository, ServerApiProvider serverApiProvider) {
+    this.client = client;
+    this.configRepository = configRepository;
+    this.connectionRepository = connectionRepository;
+    this.serverApiProvider = serverApiProvider;
+  }
+
+  @Subscribe
+  public void configurationScopesAdded(ConfigurationScopesAddedEvent event) {
+    var configScopeIds = event.getAddedConfigurationScopeIds();
+    notifyIfSoonUnsupported(configScopeIds);
+  }
+
+  @Subscribe
+  public void bindingConfigChanged(BindingConfigChangedEvent event) {
```

Comment:
When changing the binding of multiple projects in a row (quite common in SLE), the client will receive many notifications for the same connection...

Reply:
If the user mutes the first notification when doing the first update, he won't receive any after that

IMO this is expected behaviour, this event is not in a 'batch' so I cannot know right away if the events are linked

Reply:
Following the main comment that I didn't see, caching could be done but then, in case I switch from `Connection_A` -> `Connection_B` -> `Connection_A`, I would expect to see the message again for `Connection_A`, WDYT?

Reply:
What about keeping a Set of connection id for which a notification was already sent? This won't survive a restart of the backend, but at least this would avoid checking many time the same connection?

Reply:
But if the user switches to the same connection multiple times during his session, and if he didn't mute the notification on the IDE side, maybe he would like to see it multiple times?

If the responsibility of 'one-time' is moved to the core, then we should the notification one time on client side and mute it by default after that, but then he maybe not the best UX because user might miss it

Reply:
In this case the user will receive the notification the next time SonarLint starts, I don't think it is a big deal here. But do as you want.

Reply:
I switched to use the storage to retrieve the version, so it avoids making a lot of unnecessary calls to the server. The downside is that it might be unsynchronized if the server is updated but that's a very rare case and only last for an hour at most.

Then I also switch to an in-memory cache for the connection/version already seen, which is cleared at each restart.